spring-integration-core/src/main/java/org/springframework/integration/annotation/Aggregator.java

```diff
@@ -104,8 +104,16 @@
 	 * @return the {@link Poller} options for a polled endpoint
 	 * ({@link org.springframework.integration.scheduling.PollerMetadata}).
 	 * This attribute is an {@code array} just to allow an empty default (no poller).
-	 * Only one {@link Poller} element is allowed.
+	 * Mutually exclusive with {@link #reactive()}.
 	 */
 	Poller[] poller() default { };
 
+	/**
+	 * @return the {@link Reactive} options for a consumer endpoint.
+	 * This attribute is an {@code array} just to allow an empty default (not reactive).
+	 * Only one {@link Reactive} element is allowed.
+	 * Mutually exclusive with {@link #poller()}.
+	 */
+	Reactive[] reactive() default { };
```

Comment:
OK. `@Reactive("not_provided")` (or similar marker) will work. Otherwise I would argue how to determine "no reactive" then: the function is optional, but users really would like to make an endpoint reactive independently of the input channel.

Reply:
I actually don't have a strong opinion on this, given that we do the same with `@Poller` - so feel free to reject the suggestion.

Reply:
(all `@Poller` attributes have defaults too).

Reply:
Yeah... I think at the moment we introduced a `@Poller` we just didn't think about some some `not_provided` marker. 
I believe we can break it and bring it back to a single annotation in `6.0`. See `ValueConstants.DEFAULT_NONE`